gobblin-runtime/src/main/java/gobblin/runtime/JobContext.java

```diff
@@ -267,6 +272,13 @@ void storeJobExecutionInfo() {
     }
   }
 
+  @Subscribe
+  public void handleNewOutputTaskStateEvent(NewOutputTaskStateEvent newOutputTaskStateEvent) {
+    this.jobState.addTaskState(newOutputTaskStateEvent.getTaskState());
+    // Update the job execution history store upon every task completion
+    storeJobExecutionInfo();
```

Comment:
Calling this for every `newOutputTaskStateEvent`, what's the overhead? Is it possible to call this method after a certain number of `newOutputTaskStateEvent` have been received?